src/lib/routes/unleash-types.ts

```diff
@@ -1,13 +1,25 @@
 import { Request } from 'express';
-import User from '../types/user';
+import IUser from '../types/user';
+import { IApiUser } from 'lib/types';
 
 export interface IAuthRequest<
     PARAM = any,
     ResBody = any,
     ReqBody = any,
     ReqQuery = any,
 > extends Request<PARAM, ResBody, ReqBody, ReqQuery> {
-    user: User;
+    user: IUser;
+    logout: (() => void) | ((callback: (err?: any) => void) => void);
+    session: any;
+}
+
+export interface IApiRequest<
+    PARAM = any,
+    ResBody = any,
+    ReqBody = any,
+    ReqQuery = any,
+> extends Request<PARAM, ResBody, ReqBody, ReqQuery> {
+    user: IApiUser;
```

Comment:
Define another interface. The main difference is that some endpoints use the `user.id` to associate the request to a user, but tokens don't have an id associated. This is the part that I think we should look at carefully...